src/languages/coffeescript.ts

```diff
@@ -38,67 +37,58 @@ export default {
 				pattern: /@(?!\d)\w+/,
 				alias: 'variable',
 			},
-		});
-
-		insertBefore(coffeescript, 'comment', {
-			'multiline-comment': {
-				pattern: /###[\s\S]+?###/,
-				alias: 'comment',
-			},
-
-			// Block regexp can contain comments and interpolation
-			'block-regex': {
-				pattern: /\/{3}[\s\S]*?\/{3}/,
-				alias: 'regex',
-				inside: {
-					'comment': comment,
-					'interpolation': interpolation,
-				},
-			},
-		});
-
-		insertBefore(coffeescript, 'string', {
-			'inline-javascript': {
-				pattern: /`(?:\\[\s\S]|[^\\`])*`/,
-				inside: {
-					'delimiter': {
-						pattern: /^`|`$/,
-						alias: 'punctuation',
+			$insertBefore: {
+				'comment': {
+					'multiline-comment': {
+						pattern: /###[\s\S]+?###/,
+						alias: 'comment',
 					},
-					'script': {
-						pattern: /[\s\S]+/,
-						alias: 'language-javascript',
-						inside: 'javascript',
+
+					// Block regexp can contain comments and interpolation
+					'block-regex': {
+						pattern: /\/{3}[\s\S]*?\/{3}/,
+						alias: 'regex',
+						inside: {
+							'comment': comment,
+							'interpolation': interpolation,
+						},
 					},
 				},
-			},
+				'string': {
+					'inline-javascript': {
+						pattern: /`(?:\\[\s\S]|[^\\`])*`/,
+						inside: {
+							'delimiter': {
+								pattern: /^`|`$/,
+								alias: 'punctuation',
+							},
+							'script': 'javascript',
```

Comment:
Shouldn't that be something like

```js
'script': { $inside: 'javascript` }
```

Reply:
No, I think that happens automatically. Or at least it should (though I’m starting to think special casing languages as strings is not a good idea but let’s get the branch merged first before changing that)

Reply:
Okay. But what is the difference between `inside` and `$inside`? Is `inside` something that should be replaced with `$inside` eventually (like it happened with `rest` that became `$rest`)?

Reply:
Yes, we should add the `language-xxx` class automatically.